File: Task-14 (Read API store HDFS)/getChuckNorris.scala
import java.io.{File, PrintWriter}
import java.net.{HttpURLConnection, URL, URLConnection}

object getChuckNorris {
  def main(args: Array[String]): Unit = {
    val url: URL = new URL("https://api.chucknorris.io/jokes/random")

    val conn: URLConnection = url.openConnection
    conn.addRequestProperty("User-Agent", "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.74 Safari/537.36 Edg/79.0.309.43");
    print("conn: " + conn.getClass + "\n")

    val input_stream = conn.getInputStream
    val content = io.Source.fromInputStream(input_stream).mkString

    input_stream.close

    val local_path = "scalaData.json"

    val local_writer = new PrintWriter(new File(local_path))
    local_writer.write(content)
    local_writer.close
  }
}

File: Task-14 (Read API store HDFS)/getChuckNorris.py

import json
import requests

from io import StringIO


output = StringIO()


fileWriter = open("data.json", "w+")
counter = 0
stringIO = StringIO()
while counter <= 10:
    response = requests.get("https://api.chucknorris.io/jokes/random")
    stringIO.write(str(response.json())+"\n")
    print(str(response.json())+"\n")
    counter += 1


data = stringIO.getvalue()

fileWriter.write(data[0:len(data)])
stringIO.close()
fileWriter.close()

# fileWriter.write(str(response.json())+ "\n")

fileWriter.close()